In [1]:
import pandas as pd
from scipy import stats
import numpy as np

### Выберете тест и проверьте, есть ли различия между выборками:

### Задача 1:
    
Даны две независимые выборки. Не соблюдается условие нормальности

x1 380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции

In [8]:
x=np.array([380,420, 290])
y=np.array ([140,360,200,900])

При сравнении 2-ух независимых выборок используем критерий Манна-Уитни:

In [9]:
stats.mannwhitneyu(x, y)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

pvalue=0.628 > alfa 0.05, делаем Вывод: статистически значимых различий между выборками НЕТ, верна альтернативная гипотеза H1.

### Задача 2:
    
Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150, 130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

In [ ]:
При анализе повторных измерений в более, чем 2-ух выборках используем критерий Фридмана:

In [10]:
x0=np.array([150, 160, 165, 145, 155])
x10=np.array([140, 155, 150, 130, 135])
x30=np.array([130, 130, 120, 130, 125])

In [11]:
stats.friedmanchisquare(x0, x10, x30)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

pvalue=0.008 < alfa 0.05, делаем Вывод: статистически значимые различия между выборками ЕСТЬ, верна нулевая гипотеза H0.

### Задача 3:

Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.


При сравнении 2-ух зависимых выборок используем критерий Уилкоксона:

In [13]:
x1=np.array([150, 160, 165, 145, 155])
x2=np.array([140, 155, 150, 130, 135])

In [14]:
stats.wilcoxon(x1,x2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

pvalue=0.0625 > alfa 0.05, делаем Вывод: статистически значимых различий между выборками НЕТ, верна альтернативная гипотеза H1.

### Задача 4:

Даны 3 группы учеников плавания.


В 1 группе время на дистанцию 50 м составляют:

56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54


При анализе независимых выборок в более, чем 2-ух выборках используем критерий Крускала-Уоллиса:

In [17]:
Gr1=np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
Gr2=np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
Gr3=np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

In [18]:
stats.kruskal(Gr1, Gr2, Gr3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

pvalue=0.06503 > alfa 0.05, делаем Вывод: статистически значимых различий между выборками НЕТ, верна альтернативная гипотеза H1.

### Задача 5:

Заявляется, что партия изготавливается со средним арифметическим 2,5 см. 

Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. 

Объем выборки 10, уровень статистической значимости 5%


2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [ ]:
Для проверки решим задачу письменно и с помощью функции:

При анализе независимых выборок в более, чем 2-ух выборках используем критерий Крускала-Уоллиса: